tests/integration/targets/ios_l3_interface/tests/cli/deleted.yaml

```diff
@@ -0,0 +1,29 @@
+---
+- debug:
+      msg: "START ios_l3_interfaces Deleted integration tests on connection={{ ansible_connection }}"
+
+- include_tasks: reset_config.yaml
+
+- name: Delete attributes of given interfaces (Note: This won't delete the interface itself)
+  ios_l3_interfaces:
+    config:
+      - name: GigabitEthernet0/1
+      - name: GigabitEthernet0/2
+      - name: GigabitEthernet0/3
+      - name: GigabitEthernet0/3.100
+    state: deleted
+
+- ios_facts:
+    gather_subset: net_configuration_interfaces
+
+- set_fact:
+    expected_output:
+      - name: GigabitEthernet0/1
+      - name: GigabitEthernet0/2
+      - name: GigabitEthernet0/3
+      - name: GigabitEthernet0/3.100
+- assert:
+    that:
+      - "ansible_facts.net_configuration.l3_interfaces == expected_output"
```

Comment:
We need to assert `before`, `after` and list of commands generated. The ordering of the before/after list of dictionaries are not consistent across consecutive runs in different environments, which may result in integration test failures in Zuul (as observed with VyOS tests in Zuul). Since we don't have an identifier for the dicts generated, we need to convert them to a set and then assert (using `symmetric_difference`), thereby making sure that changes in ordering does not affect test results.